WIP - nestable BREAD ordering

MrCrayon opened the pull request:
Make ordering nestable if Trait is used by Model, Model will need to set parent id field like:
```php
use TCG\Voyager\Traits\Nestable;

class Category extends Model
{
    use Nestable;

    protected $parentField = 'parent_id';
```

Fixes order not mantained while using descending order direction.

Closes #4641

Co-Authored-By: dzianisyaukhuta <dzianisyaukhuta@users.noreply.github.com>

MrCrayon committed:
WIP - nestable BREAD ordering

Make ordering nestable if Trait is used by Model, Model will need to set parent id field like:
`protected $parentField = 'parent_id';`

Fixes order not mantained while using descending order direction.

Closes #4641

Co-Authored-By: dzianisyaukhuta <dzianisyaukhuta@users.noreply.github.com>

MrCrayon committed:
StyleCI fixes

Co-Authored-By: dzianisyaukhuta <dzianisyaukhuta@users.noreply.github.com>

fletch3555 commented on resources/views/bread/order.blade.php:
```diff
@@ -50,7 +34,7 @@
 $(document).ready(function () {
     $('.dd').nestable({
-        maxDepth: 1
+        @if(empty($nestable)) maxDepth: 1 @endif
```
Why check `empty()` here when you return a boolean from the controller?

MrCrayon replied:
I'm trying to avoid problems in case someone already overrode order method and updates.
Following that logic I forgot to do the same in the partial view. 
Let me know if you prefer instead that we force the user to keep controller updated.

fletch3555 commented on resources/views/bread/partials/order.blade.php:
```diff
@@ -0,0 +1,23 @@
+                <span>{{ $result->{$display_column} }}</span>
+            @endif
+        </div>
+        @if(isset($result->nestableChildren) && !$result->nestableChildren->isEmpty())
```
So.. nestableChildren holds the child entries... do you need $testable at all? You could just check if this list is empty, right?

MrCrayon replied:
I can only check if element is under the right parent if it's nestable, otherwise there is no parent id. 

I need to change parent_id that's the hard coded value I was using at start.

MrCrayon committed:
Trait

Forgot to include Trait
Changed hardcoded parent_id
Always use empty($nestable)

dzianisyaukhuta commented:
Thank you so much @MrCrayon !!!  I hope we still get a sort of BREAD in a tree view.

MrCrayon commented:
@dzianisyaukhuta the result is the same as your PR but with much less code :grin: 
![Schermata da 2020-01-10 10-58-26](https://user-images.githubusercontent.com/6875243/72144216-776e8a00-3398-11ea-9f3a-16fe66d30e8d.png)

dzianisyaukhuta commented:
> @dzianisyaukhuta the result is the same as your PR but with much less code 
> ![Schermata da 2020-01-10 10-58-26](https://user-images.githubusercontent.com/6875243/72144216-776e8a00-3398-11ea-9f3a-16fe66d30e8d.png)

Bravo!

dzianisyaukhuta commented:
Can anybody approve it? @fletch3555

dzianisyaukhuta commented:
Why?

fletch3555 commented:
Why what? Why is it not approved/merged? That answer is simple. It's still labeled `wip`

dzianisyaukhuta commented:
@MrCrayon big respect to you. You are awesome.

cotiga commented:
I added this to a fork under 1.4.2, works great.
Please add this kind of functionality.
Voyager seems to stagnate lately, please don't give up on us

emptynick commented:
Closing as this will be implemented in Voyager 2
